### ISSUE TITLE
Autogenerate player ids

### ISSUE THREAD

**Issue by czarandy** [czarandy/mtgptresults#24]
Having player ids be manually set causes some problems: it's easy to make typos and we are not super consistent with how we generate them from names. We should change it to auto-generate the ids from the names. We just need to decide on the rules for doing so.

**Comment by OdinFK** [comment]
What do we need to do more specifically? I think the following steps are necessary

For every player entry in a given tournament file
(1) Generate an 'ID' of the 'player name' and replace the current 'ID' with the new one
(2) Copy the 'ID' and 'player name' into the playerdata.json
(3) Remove the 'player name' from the entry in the tournament file
(4) Return a list of the added players to the console

Comments
Step 1: If at all possible I would like to have the IDs be generated in a way similar to what they are now. Numeric IDs are a possibility, but this will make editing the tournament files way more inconvenient. I don't know if there is a table that maps all special characters from the Latin alphabet to the Latin standard alphabet, but I suppose there must be such a thing. I would generate the IDs from the name by replacing all special letters with their standard pendant, and remove all special characters and periods. Of course this process will lead to the occasional duplicate, but so will creating a hash ID from identical names.

Step 2: Obvious I think. Name is provisional of course.

Step 3: There is no need to keep the names in the individual tournament files. It just inflates the tournament files. The name is a natural part to the player data file.

Step 4: This is not strictly necessary, but I believe it would be useful when adding new tournaments in the future. Having all newly added names in a list gives us a convenient way to check these entries for correctness.

**Comment by mlnmln** [comment]
> (1) Generate an 'ID' of the 'player name' and replace the current 'ID' with the new one

We should be able to use st. like https://github.com/FGRibreau/node-unidecode to generate passable ids as @czarandy has stated. I think ids should be as human-readable as possible, as we also use them as part of the players route on the site. 

> (2) Copy the 'ID' and 'player name' into the playerdata.json

Not really sure how I feel about this. The player id is a derivative from the player's name. And I don't like having duplicated data in our repository. If the `playerdata.json` would be just part of the build, I'd be fine with that. 

> (3) Remove the 'player name' from the entry in the tournament file

We have to think about how we include future tournaments. If we replace the name by the id we would need additional steps when running the standings parser. 

> (4) Return a list of the added players to the console

Agree that this would be good, but I foresee some difficulties @czarandy How could we get a reliable diff without storing the playerdata.json in the repository?

**Comment by czarandy** [comment]
Actually I thought we could keep just the *names* in the tournament json files. The ids would be autogenerated based on the names. The advantage is that it's easier to add new tournaments (since wotc is just giving you the name in their standings) and we avoid any duplication of data. 

For extra player data we can still use the names and have a file that includes that information keyed on name, then the build step will also convert it to be keyed by id.

If you want to check for new names in a tournament, we could just have a separate script that does that.

**Comment by OdinFK** [comment]
My reasoning behind having an ID in the tournament file, and the name in the playerdata file was the following. Take Martin Juza for example. Currently we write him Martin Juza. Suppose we want to change his name to Martin Jůza (native spelling). Wouldn't it be much more complicated to change that? Do I have it wrong, or would we have to change Martin Juza in all tournament file in your version?

**Comment by czarandy** [comment]
It's easy enough to rename in all the files, like this:
  `find . -type f -exec sed -i "s/Martin Juza/Martin Jůza/g" {} \;`

**Comment by mlnmln** [comment]
@czarandy We should respect that not everyone is comfortable in the command line. But I second that doing search and replace over multiple files is as trivial doing only one file.

**Comment by mlnmln** [comment]
@czarandy Who is going to work on this? You or me?

**Comment by czarandy** [comment]
I will do it

**Comment by czarandy** [comment]
Ok so I made it generate ids from player names. the json files no longer have ids in them. This changed a few ids and thus the urls, but they should all be better, e.g.: http://www.mtgptresults.com/player/raphael-levy

I skimmed over all the changes and they seemed fine. The main benefit is that now it's a lot easier to add players/tournaments. I can now add the missing players from the first three pts

I didn't change the format of the generated files at all. It's possible that removing some of the duplication in the files would make them a bit smaller, but it didn't seem too important (and it might not matter that much after gzip anyway).
